### ISSUE TITLE
Feature: Throw an error if model class has multiple foreign keys referencing the same model

### ISSUE THREAD

**Issue by RobinBuschmann** [RobinBuschmann/sequelize-typescript#401]
sequelize-typescript should throw an error if it finds multiple foreign keys referencing the same model like:

```ts
@Table
export class FriendList extends Model<FriendList> {

  @ForeignKey(() => User)
  @Column
  UserId: number;

  @ForeignKey(() => User)
  @Column
  FriendId: number;
}
```
This should be because it cannot infer which is the actual `foreignKey` and which one is the `otherKey`.

**Comment by damianb** [comment]
This would be very, very valuable on more complex schemas.

Silently ignoring the other key would not have gone noticed if I wasn't looking directly at the generated schema just now, and it took me some digging to remember *why* this was a problem.  This is an error case and warrants throwing an error as such.
